### ISSUE TITLE
Scraper regex only works for filename rather than directory path - issue when building Fido clients

### ISSUE THREAD

**Issue by hayesla** [sunpy/sunpy#6924]
### Describe the bug

When using `sunpy.net.scraper.Scraper`, and/or extending the Fido `GenericClient`, the regex pattern approach does not work unless for the url path, it only works for the filename.

This is an issue that becomes clear when 

For example:
Here the GOES-XRS file resolution is also named within the file directory and also filename, namely `../data/xrsf-l2-flx1s_science/yyyy/mm/sci_xrsf-l2-flx1s_g16_dyyyymmdd_v2-1-0.nc.`, where here the string `flx1s` is the defining keyword. The 1 minute averages are at the same paths except with `avg1m`. Anywho, this is just an example to show this
```python
>>> baseurl_r = (r"https://data.ngdc.noaa.gov/platforms/solar-space-observing-satellites/goes/goes16"
             r"/l2/data/xrsf-l2-(\w){5}_science/%Y/%m/sci_xrsf-l2-(\w){5}_g16_d%Y%m%d_.*\.nc")
>>> scraper_goes = Scraper(baseurl_r, regex=True)
>>> scraper_goes.filelist(TimeRange("2022-02-15 01:00", "2022-02-16"))
[]
```
now if we fix the url path but use regex for just the filename it works
```python
>>> baseurl_r = (r"https://data.ngdc.noaa.gov/platforms/solar-space-observing-satellites/goes/goes16"
                 r"/l2/data/xrsf-l2-flx1s_science/%Y/%m/sci_xrsf-l2-(\w){5}_g16_d%Y%m%d_.*\.nc")
>>> scraper_goes = Scraper(baseurl_r, regex=True)
>>> scraper_goes.filelist(TimeRange("2022-02-15 01:00", "2022-02-16"))
['https://data.ngdc.noaa.gov/platforms/solar-space-observing-satellites/goes/goes16/l2/data/xrsf-l2-flx1s_science/2022/02/sci_xrsf-l2-flx1s_g16_d20220215_v2-1-0.nc',
 'https://data.ngdc.noaa.gov/platforms/solar-space-observing-satellites/goes/goes16/l2/data/xrsf-l2-flx1s_science/2022/02/sci_xrsf-l2-flx1s_g16_d20220216_v2-1-0.nc']
```
So basically it just needs to be in the docs for scraper, and indeed thought about for the scraper rewrite

### To Reproduce

import sunpy

sunpy.map.Map(...)
etc

**Comment by hayesla** [comment]
this is also relevant and mentioned in #4336

**Comment by nabobalis** [comment]
NOTE: Check if this is closed by https://github.com/sunpy/sunpy/pull/7077
